connector_alndata/models/crm_lead.py

```diff
@@ -0,0 +1,708 @@
+    'C': 'US/Central',
+    'E': 'US/Eastern',
+    'P': 'US/Pacific',
+    'H': 'US/Hawaii',
+}
+
+
+class Lead(models.Model):
+    _inherit = "crm.lead"
+
+    lastdate_changed = fields.Datetime('LastDateChanged')
+    industry_id = fields.Many2one('res.partner.industry', 'Market')
+    lead_type = fields.Selection(
+        [('owner', 'Owner'),
+         ('management_company', 'Management Company'),
+         ('contact', 'Contact'),
+         ('new_construction', 'New Constructions')],
+        'Lead Type')
+
+    def aln_auth_login(self, data_key='', data_params=None):
+        """Aln Connector.
+
+        This method is used to connect with ALN and fetch the data.
+        """
+        config_obj = self.env['ir.config_parameter']
+        url = config_obj.get_param('alndata.api.url')
+        api_key = config_obj.get_param('alndata.api.key')
+        full_content = []
+
+        url = url + data_key
+        count = 0
+        flag = True
+        params = {
+            'apikey': api_key,
+            'Accept': 'application/json',
+        }
+        _logger.info('========== %s ==========', data_key)
+        if not data_params:
+            data_params = {}
+        params.update(data_params)
+
+        # use count to fetch all data related filter by multiple request
+        while flag:
+            try:
+                if count > 0:
+                    params.update({'$skip': count})
+                response = requests.get(url, params=params)
+                response.raise_for_status()
+            except Exception as e:
+                _logger.error('%s', e)
+            content = response.content.decode('utf8')
+            if content:
+                content = json.loads(content).get('value')
+                if len(content or []) > 0:
+                    count += len(content)
+                    full_content += content
+                else:
+                    flag = False
+                    count = 0
+        _logger.info('========== No of datas : %s ==========',
+                     len(full_content))
+        return full_content or []
+
+    def remove_data(self, model, datas, origin=''):
+        """Removed Data.
+
+        This method is used to removed data from odoo database.
+        which deleted from ALN Data.
+        """
+        obj = self.env[model]
+        removed_ids = []
+        rem = obj
+        domain = []
+        if model == 'res.partner':
+            domain += [('partner_type', '=', origin)]
+        elif origin and model == 'crm.lead':
+            domain += [('lead_type', '=', origin)]
+        for rec in obj.search(domain):
+            if model == 'crm.lead' and rec.referred not in datas:
+                removed_ids.append(rec.id)
+                rem += rec
+            elif model in ['res.partner', 'fsm.location'] and \
+                    rec.ref not in datas:
+                removed_ids.append(rec.id)
+                rem += rec
+        rem.unlink()
+        return removed_ids
```

Comment:
@nikitavaghela I would not delete data. We may have transactions with those records. Simply archive the records.

Reply:
okay, I will update it.